### DISCUSSION TITLE
Statistiken für Float/Boolean-SysVar-Entitäten aktivieren... machbar/sinnvoll?

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#704] [Q&A]
Die neuen Statistik Funktionen in HA finde ich sehr praktisch.

Als "langsamer Umsteiger" nutze ich auch einige wenige Systemvariablen der Zentrale in HA.
Nicht unbedingt für die Automatisierung sondern eher im Dashboard für den schnellen "Kontrollblick".
(Beispielhaft hier Temperatur/RAM-Usage/Auslastung des Zentralen-Host's (Pi3B+))

Primär würde ich mir die aktivierte Statistik für Float-SysVars (Typ: Zahl) wünschen, ob ich das auch für Boolean-SysVars (Typ: Logik) gebrauchen könnte weiß ich gerade nicht.

Wäre das machbar und auch sinnvoll?

**Comment by SukramJ** [comment]
Grundsätzlich ist das für alle numerischen Sensoren machbar, also Sensor Entitäten vom CCU Typ Zahlenwert.
Wir durch die Verwendung von `hahm` in der Beschreibung eine Number Entität erstellt gibt es derzeit keine Langzeitstatistiken.

Aber:
Ganz so einfach ist das mit den Langzeitstatistiken nicht, wie auch aus der [HA Dev Doku](https://developers.home-assistant.io/docs/core/entity/sensor#long-term-statistics) und [hier](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes) zu entnehmen ist. Viel Spass beim lesen.

Die meisten numerischen Sensoren würde wohl als `state_class` `measurement` angelegt  werden, aber da es auch `total` und `total_increasing` Fälle gibt, wird die codeseitige Unterscheidung schwierig.

Ich würde das aktuell über die `customize.yml` lösen, und die entsprechende `state_class` setzen.

**Reply by Baxxy13** [reply]
Alles klar und danke für die Links.
Habe mich da jetzt mal belesen und komme nun auch zu dem Schluss das das über die `customize.yml` am einfachsten zu handhaben ist.

Da du ja nicht wissen kannst was der User da so für Werte in seinen SysVars hat, kannst du natürlich auch die `state_class` (bei SysVar Typ: Zahl) nicht definieren. Da fehlten mir bei der Idee noch die relevanten Informationen. 

Bei Logik-SysVar (BINARY SOURCE SENSOR) könnte man das aber problemlos aktivieren, oder? Hier ist der Status ja "definiert".

**Reply by SukramJ** [reply]
Langzeitstatistiken gibt es nur für Sensoren.
